scipy/fft/_fftlog.py

```diff
@@ -242,37 +247,39 @@ def ifht(A, dln, mu, offset=0.0, bias=0.0):
     :math:`\mu` may be any real number, positive or negative.
 
     See `fht` for further details.
+    """
+    xp = array_namespace(A)
 
-    '''
+    if not from_uarray and is_numpy(xp):
+        return copy(_fftlog_uarray.ifht(A, dln, mu, offset=offset, bias=bias,
+                                        from_uarray=True))
 
     # size of transform
-    n = np.shape(A)[-1]
+    n = A.shape[-1]
 
     # bias input array
     if bias != 0:
         # A_q(k) = A(k) (k/k_c)^{q} (k_c r_c)^{q}
         j_c = (n-1)/2
-        j = np.arange(n)
-        A = A * np.exp(bias*((j - j_c)*dln + offset))
+        j = xp.arange(n, dtype=xp.float64)
+        A = A * xp.exp(bias*((j - j_c)*dln + offset))
 
     # compute FHT coefficients
-    u = fhtcoeff(n, dln, mu, offset=offset, bias=bias)
+    u = xp.asarray(fhtcoeff(n, dln, mu, offset=offset, bias=bias))
 
     # transform
-    a = _fhtq(A, u, inverse=True)
+    a = _fhtq(A, u, inverse=True, xp=xp)
 
     # bias output array
     if bias != 0:
         # a(r) = a_q(r) (r/r_c)^{q}
-        a /= np.exp(-bias*(j - j_c)*dln)
+        a /= xp.exp(-bias*(j - j_c)*dln)
 
-    return a
+    return copy(a)
```

Comment:
Same here, why do we need a `copy`? We're only using `xp.<func>` calls, so the array should be of the correct type already.